src/mlpack/core/data/dataset_mapper_impl.hpp

```diff
@@ -81,20 +81,54 @@ inline T DatasetMapper<PolicyType>::MapString(const std::string& string,
 
 // Return the string corresponding to a value in a given dimension.
 template<typename PolicyType>
+template<typename T>
 inline const std::string& DatasetMapper<PolicyType>::UnmapString(
-    const size_t value,
-    const size_t dimension)
+    const T value,
+    const size_t dimension,
+    const size_t unmappingIndex)
 {
+  // If the value is std::numeric_limits<T>::quiet_NaN(), we can't use it as a
+  // key---so we will use something else...
+  const T usedValue = (value != value) ?
```

Comment:
Currently we don't use ``-ffast-math`` so we don't run into any problems here: 
https://stackoverflow.com/questions/22931147/stdisinf-does-not-work-with-ffast-math-how-to-check-for-infinity

but maybe it's worth to take a closer look into `-ffast-math`` and to think about a workaround? Also this issue was raised by the static code analysis check.